iotlabsshcli/tests/open_a8_ssh_test.py

```diff
@@ -95,95 +107,76 @@ def test_scp(connect, client, put, _open):
 
     assert ret == {'0': ['saclay.iot-lab.info', 'grenoble.iot-lab.info']}
 
+    # pylint: disable=R0204
+    connect.side_effect = ConnectionErrorException()
+    ret = node_ssh.scp(src, dst)
+
+    assert ret == {'1': ['saclay.iot-lab.info', 'grenoble.iot-lab.info']}
+
     # Simulating an exception
     connect.side_effect = AuthenticationException()
 
     with raises(OpenA8SshAuthenticationException):
         node_ssh.scp(src, dst)
 
 
-@patch('pssh.SSHClient')
-@patch('pssh.SSHClient._connect_tunnel')
-def test_wait_all_boot(connect, client, capsys):
+@patch('pssh.ParallelSSHClient.run_command')
+@patch('pssh.ParallelSSHClient.join')
+def test_wait_all_boot(join, run_command):
     # pylint: disable=unused-argument
     """Test wait for ssh nodes to be available."""
     config_ssh = {
         'user': 'username',
         'exp_id': 123,
     }
+
+    test_command = 'test'
     groups = _nodes_grouped(_ROOT_NODES)
 
     # normal boot
     node_ssh = OpenA8Ssh(config_ssh, groups, verbose=True)
-    ret = node_ssh.wait(120)
-    out, _ = capsys.readouterr()
-
-    assert len(out.split('\n')) == len(_ROOT_NODES) + 1
 
-    assert ret == {'0': _nodes_from_groups(groups)}
-    assert '1' not in ret.keys()
-
-
-@patch('pssh.SSHClient')
-@patch('pssh.SSHClient._connect_tunnel')
-def test_wait_authentication_exc(connect, client, capsys):
-    # pylint: disable=unused-argument
-    """Test wait for ssh nodes to be available."""
-    config_ssh = {
-        'user': 'username',
-        'exp_id': 123,
-    }
-    groups = _nodes_grouped(_ROOT_NODES)
+    # Print output of run_command
+    run_command.return_value = dict(
+        (node,
+         {'stdout': ['test'], 'exit_code': 0})
+        for node in _ROOT_NODES)
 
-    node_ssh = OpenA8Ssh(config_ssh, groups, verbose=True)
-    # Simulating an authentication exception
-    connect.side_effect = AuthenticationException()
+    node_ssh.wait(120)
+    node_ssh.run(test_command)
 
-    with raises(OpenA8SshAuthenticationException):
-        node_ssh.wait(120)
+    run_command.call_count = len(_ROOT_NODES)
+    run_command.assert_called_with(test_command,
```

Comment:
The real problem here is that `run_command.call_count = len(_ROOT_NODES)` does not test anything.